### DISCUSSION TITLE
佬们帮我看看我这post要传参的api就说不存在，用get就没问题

### DISCUSSION THREAD

**Discussion by Absvion** [Mrs4s/go-cqhttp#1948] [Q&A] [answered]
![s](https://user-images.githubusercontent.com/110546817/222751031-cbb7707e-6e92-4280-bc95-d0e8900c0a55.png)

我把group_id字段弄成字符串（ bodyMap.put("group_id",”906795766”);）也返回一样的。只要不传参数就没问题=-=  求佬们救救

`import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.HashMap;
import java.util.Map;
import com.alibaba.fastjson.JSON;

public class Dopost {
	

	public static String sendPost(String uri, String charset,Map<String,Object> bodyMap, 
    Map<String,String> headerMap) {
        String result = null;
        OutputStreamWriter out = null;
        InputStream in = null;
        try {
            URL url = new URL(uri);
            HttpURLConnection urlcon = (HttpURLConnection) url.openConnection();
            urlcon.setDoInput(true);
            urlcon.setDoOutput(true);
            urlcon.setUseCaches(false);
            urlcon.setRequestMethod("POST");
            if (!headerMap.isEmpty()){
                for (Map.Entry<String, String> entry : headerMap.entrySet()) {
                    urlcon.setRequestProperty(entry.getKey(),entry.getValue());
                }
            }
            // 获取连接
            urlcon.connect();
            out = new OutputStreamWriter(urlcon.getOutputStream(),"UTF-8");
            //请求体里的内容转成json用输出流发送到目标地址
            out.write(JSON.toJSONString(bodyMap).toString());
            System.out.println("发送："+JSON.toJSONString(bodyMap).toString());
            out.flush();
            in = urlcon.getInputStream();
            BufferedReader buffer = new BufferedReader(new InputStreamReader(in, charset));
            StringBuffer bs = new StringBuffer();
            String line = null;
            while ((line = buffer.readLine()) != null) {
                bs.append(line);
            }
            result = bs.toString();
            System.out.println("接收："+result);
        } catch (Exception e) {
            System.out.println("[请求异常][地址：" + uri + "][错误信息："+ e.getMessage() + "]");
        } finally {
            try {
                if (null != in){
                    in.close();
                }
                if (null != out){
                    out.close();
                }
            } catch (Exception e2) {
                System.out.println("[关闭流异常][错误信息：" + e2.getMessage() + "]");
            }
        }
        return result;
    }
	
      public static  void main(String[] args) throws IOException {
    	  
    	  
    	  String uri="http://127.0.0.1:44857/send_group_msg";
          String charset="UTF-8";
          //请求体设置
          
          HashMap<String, Object> bodyMap = new HashMap<>();
          bodyMap.put("group_id",906795766L);
          bodyMap.put("message","testtest");
          
          //请求头设置
          HashMap<String, String> headerMap = new HashMap<>();
          headerMap.put("Content-Ttype","application/json");
          //调用
          sendPost(uri,charset,bodyMap,headerMap);

          //SimpleDateFormat df = new SimpleDateFormat("yyyyMMddHHmmssSSS");//设置日期格式
          //String date = df.format(new Date());// new Date()为获取当前系统时间，也可使用当前时间戳
    	  //System.out.println(date);

    	  
  

      }
    	  
}`

**Comment by BobLiu0518** [comment, marked as answer]
会不会是 Content-type 误写成 Content-Ttype 的原因（

**Reply by Absvion** [reply]
眼瞎了，就是这个问题了(
谢谢佬
